cmd/erigon-el/stages/stages.go

```diff
@@ -48,7 +48,7 @@ func NewStagedSync(ctx context.Context,
 	if cfg.Snapshot.Enabled {
 		blockReader = snapshotsync.NewBlockReaderWithSnapshots(snapshots)
 	} else {
-		blockReader = snapshotsync.NewBlockReader()
+		blockReader = snapshotsync.NewBlockReader(cfg.TransactionsV3)
```

Comment:
Let's try to remove `snapshotsync.NewBlockReader(` and always use `snapshotsync.NewBlockReaderWithSnapshots(` even when snapshots are disabled (it anyway works when "no snapshots created yet").

Reply:
Okay will do!

Reply:
https://github.com/ledgerwatch/erigon/blob/devel/eth/stagedsync/stage_mining_exec.go#L76
This is the only place I was not able how to replace with `snapshotsync.NewBlockReaderWithSnapshots(` to pass the `*RoSnapshots`

Reply:
After this one, I will also remove the class

Reply:
Done! Please check at your convenience.